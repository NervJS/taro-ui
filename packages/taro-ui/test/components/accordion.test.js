import React from 'react'
import { mount } from 'enzyme'
import AtAccordion from '../../lib/components/accordion'

const iconObject = {
  prefixClass: 'prefix',
  value: 'chevron-down',
  color: 'red'
}
const iconStyle = { color: 'red', fontSize: 'undefinedpx' } // TODO: 确保不出现 undefinedpx

describe('Accordion Component', () => {
  it('render initial Accordion', () => {
    const component = mount(<AtAccordion />)
    expect(component.find('taro-view-core')).toHaveLength(8)
    expect(component.find('taro-text-core')).toHaveLength(1)
    expect(component.html()).toMatchSnapshot()
  })

  it('render Accordion -- props title', () => {
    const component = mount(<AtAccordion title='title' />)
    expect(component.prop('title')).toBe('title')
    expect(component.html()).toMatchSnapshot()
  })

  it('render Accordion -- props open', () => {
    const component = mount(<AtAccordion open />)
    expect(
      component.find('.at-accordion__arrow--folded').length
    ).toBeGreaterThanOrEqual(1)
    expect(component.html()).toMatchSnapshot()
  })

  it('render Accordion -- props hasBorder', () => {
    const component = mount(<AtAccordion hasBorder={false} />)
    expect(
      component.find('.at-accordion__header--noborder').length
    ).toBeGreaterThanOrEqual(1)
    expect(component.html()).toMatchSnapshot()
  })

  it('render Accordion -- props icon', () => {
    const component = mount(<AtAccordion icon={iconObject} />)
    expect(component.props().icon).toEqual(iconObject)
    expect(
      component.find('.prefix.prefix-chevron-down').length
    ).toBeGreaterThanOrEqual(1)
    expect(
      component.find('.prefix.prefix-chevron-down').first().prop('className')
    ).toBe('prefix prefix-chevron-down at-accordion__icon')
    expect(
      component.find('.prefix.prefix-chevron-down').first().prop('style')
    ).toEqual(iconStyle)
    expect(component).toMatchSnapshot()
  })

  it('render Accordion -- props note', () => {
    const component = mount(<AtAccordion note='note' />)
    expect(component.prop('note')).toBe('note')
    expect(component).toMatchSnapshot()
  })
})

describe('Accordion Behavior', () => {
  it('handle Accordion -- event onClick', () => {
    const handleClick = jest.fn()
    const component = mount(<AtAccordion onClick={handleClick} />)
    component.instance().handleClick()
    expect(handleClick).toBeCalled()
  })
})
