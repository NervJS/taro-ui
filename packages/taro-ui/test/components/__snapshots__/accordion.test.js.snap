// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component render Accordion -- props hasBorder 1`] = `"<taro-view-core class=\\"at-accordion\\" style=\\"\\"><taro-view-core class=\\"at-accordion__header at-accordion__header--noborder\\"><taro-view-core class=\\"at-accordion__info\\"><taro-view-core class=\\"at-accordion__info__title\\"></taro-view-core><taro-view-core class=\\"at-accordion__info__note\\"></taro-view-core></taro-view-core><taro-view-core class=\\"at-accordion__arrow\\"><taro-text-core class=\\"at-icon at-icon-chevron-down\\"></taro-text-core></taro-view-core></taro-view-core><taro-view-core class=\\"at-accordion__content at-accordion__content--inactive\\"><taro-view-core class=\\"at-accordion__body\\"></taro-view-core></taro-view-core></taro-view-core>"`;

exports[`Accordion Component render Accordion -- props icon 1`] = `ReactWrapper {}`;

exports[`Accordion Component render Accordion -- props note 1`] = `ReactWrapper {}`;

exports[`Accordion Component render Accordion -- props open 1`] = `"<taro-view-core class=\\"at-accordion\\" style=\\"\\"><taro-view-core class=\\"at-accordion__header\\"><taro-view-core class=\\"at-accordion__info\\"><taro-view-core class=\\"at-accordion__info__title\\"></taro-view-core><taro-view-core class=\\"at-accordion__info__note\\"></taro-view-core></taro-view-core><taro-view-core class=\\"at-accordion__arrow at-accordion__arrow--folded\\"><taro-text-core class=\\"at-icon at-icon-chevron-down\\"></taro-text-core></taro-view-core></taro-view-core><taro-view-core class=\\"at-accordion__content\\"><taro-view-core class=\\"at-accordion__body\\"></taro-view-core></taro-view-core></taro-view-core>"`;

exports[`Accordion Component render Accordion -- props title 1`] = `"<taro-view-core class=\\"at-accordion\\" style=\\"\\"><taro-view-core class=\\"at-accordion__header\\"><taro-view-core class=\\"at-accordion__info\\"><taro-view-core class=\\"at-accordion__info__title\\">title</taro-view-core><taro-view-core class=\\"at-accordion__info__note\\"></taro-view-core></taro-view-core><taro-view-core class=\\"at-accordion__arrow\\"><taro-text-core class=\\"at-icon at-icon-chevron-down\\"></taro-text-core></taro-view-core></taro-view-core><taro-view-core class=\\"at-accordion__content at-accordion__content--inactive\\"><taro-view-core class=\\"at-accordion__body\\"></taro-view-core></taro-view-core></taro-view-core>"`;

exports[`Accordion Component render initial Accordion 1`] = `"<taro-view-core class=\\"at-accordion\\" style=\\"\\"><taro-view-core class=\\"at-accordion__header\\"><taro-view-core class=\\"at-accordion__info\\"><taro-view-core class=\\"at-accordion__info__title\\"></taro-view-core><taro-view-core class=\\"at-accordion__info__note\\"></taro-view-core></taro-view-core><taro-view-core class=\\"at-accordion__arrow\\"><taro-text-core class=\\"at-icon at-icon-chevron-down\\"></taro-text-core></taro-view-core></taro-view-core><taro-view-core class=\\"at-accordion__content at-accordion__content--inactive\\"><taro-view-core class=\\"at-accordion__body\\"></taro-view-core></taro-view-core></taro-view-core>"`;
